lib/storage/providers/SQLiteStorage.js

```diff
@@ -0,0 +1,106 @@
+/**
+ * The SQLiteStorage provider stores everything in a key/value store by
+ * converting the value to a JSON string
+ */
+import {QuickSQLite} from 'react-native-quick-sqlite';
+import _ from 'underscore';
+
+const DB_NAME = 'Expensify-new-db';
+QuickSQLite.open(DB_NAME);
+
+QuickSQLite.execute(DB_NAME, 'CREATE TABLE IF NOT EXISTS magic_map (record_key TEXT NOT NULL PRIMARY KEY , valueJSON JSON NOT NULL) WITHOUT ROWID;');
+QuickSQLite.execute(DB_NAME, 'PRAGMA CACHE_SIZE=-20000;');
+QuickSQLite.execute(DB_NAME, 'PRAGMA SYNCHRONOUS=NORMAL;');
+QuickSQLite.execute(DB_NAME, 'PRAGMA journal_mode=WAL;');
```

Comment:
Should we leave more information about these Pragmas here and why they were chosen these values? I know this was mostly from the call with SQLite and testing, but still some context would be handy.

Reply:
Also, did you ever try using `WAL2`? We can always do a follow-up to explore and compare benchmarks, but as I understand it can result in better performance when multiple threads are involved, particularly in the area of memory usage (preventing certain undesirable scenarios like an infinitely-growing WAL)

Reply:
@roryabraham I haven't tested it. That most likely would require to replace SQLite code that quick-sqlite uses with the version that supports WAL2. I happy to test it if you want :)

Reply:
> That most likely would require to replace SQLite code that quick-sqlite uses with the version that supports WAL2. I happy to test it if you want

Let's test out WAL2 in a follow-up